### PR TITLE
Update all code use to GA System.ClientModel 1.1.0

### PR DESCRIPTION
**Pull request by ralph-msft**

### CODE REVIEW

File: .dotnet.azure/eng/Packages.Data.props
Comment (annelo-msft) [inline .dotnet.azure/eng/Packages.Data.props:89]: 🎉

File: .dotnet.azure/sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClient.cs
Comment (annelo-msft) [inline .dotnet.azure/sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClient.cs:125]: nit: IDK if it matters or if it the right place to do this, but in the arch board yesterday IIRC they asked to remove this constructor so you could consider doing that here.  Also, a follow-up PR might be fine, I don't feel strongly.

File: .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj
Comment (annelo-msft) [inline .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj:21]: If you updated to 1.1.0 in Packages.Data.Props, I think you no longer need the version override?

File: .scripts/Run-Checks.ps1
Comment (joseharriaga) [inline .scripts/Run-Checks.ps1:83]: Could you also update the version of System.ClientModel that is referenced in the .scripts/Export-API.ps1 script here:
🔗 https://github.com/joseharriaga/openai-in-typespec/blob/main/.scripts/Export-API.ps1#L23

"1.1.0-beta.5" --> "1.1.0"

File: .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj
Comment (annelo-msft) [inline .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj:23]: Do we need this anymore?

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (annelo-msft) [inline .dotnet/api/OpenAI.netstandard2.0.cs:7]: IDK if architects will be happy with this param name - it doesn't need to block this PR but we should get their eyes on it before GA.

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (joseharriaga) [inline .dotnet/api/OpenAI.netstandard2.0.cs:8]: @annelo-msft : Did you originally add this temporarily to avoid having to fix all the code that was using the implicit cast of ApiKeyCredential that was removed, and we should remove it before merging this PR? Or did you intent to add this API?

File: .dotnet/src/Custom/Assistants/AssistantClient.Convenience.cs
Comment (annelo-msft) [inline .dotnet/src/Custom/Assistants/AssistantClient.Convenience.cs:310]: TY for updating these!

File: .dotnet/src/Custom/Assistants/AssistantClient.cs
Comment (annelo-msft) [inline .dotnet/src/Custom/Assistants/AssistantClient.cs:1203]: Looks like I had missed one in the try-cast/throw pattern update - would you mind updating this one to conform to the pattern used in others of these methods?

File: .dotnet/tests/Assistants/AssistantTests.cs
Comment (annelo-msft) [inline .dotnet/tests/Assistants/AssistantTests.cs:1595]: Should we add this one back?

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (annelo-msft) [reply .dotnet/api/OpenAI.netstandard2.0.cs:8]: Both.  IIRC we had discussed with architects that we would do this when we had removed the implicit cast so that users wouldn't have to `new ApiKeyCredential` everywhere per usability.  But if you don't want to do that, or want to address in a separate PR, we can do it differently in this PR.

**Review by annelo-msft — APPROVED** [review]
Left a few non-blocking comments and I will trust you on the AOAI test framework pieces, but the rest LGTM

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (joseharriaga) [inline .dotnet/api/OpenAI.netstandard2.0.cs:1287]: BatchClient is missing the new constructors that take a `string apiKey` parameter.

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (joseharriaga) [inline .dotnet/api/OpenAI.netstandard2.0.cs:1934]: FineTuningClient is missing the new constructors that take a `string apiKey` parameter.

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (joseharriaga) [inline .dotnet/api/OpenAI.netstandard2.0.cs:55]: BatchClient, FineTuningClient, ModelClient, ModerationClient, VectorStoreClient

**Review by joseharriaga — APPROVED** [review]

File: .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj
Comment (ralph-msft) [reply .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj:21]: Unfortunately, the way the automatic package references are different for what is considered a client library vs what is considered test code. For System.ClientModel, we currently have a package reference version for client libraries, but none for test code. Since this is test code, I added the override here so it continues to compile.

At some point, I think the package reference for System.ClientModel should be added for test code in Packages.Data.props, but since that could affects things globally, I've avoided that for now.

File: .dotnet.azure/sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClient.cs
Comment (ralph-msft) [reply .dotnet.azure/sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClient.cs:125]: Given the size of this PR right now, I want to keep it laser focused on just the minimum updates needed for code to compile and run properly with System.ClientModel 1.1.0. I think that any API changes unrelated to SCM 1.1.0 should go in a separate PR.

File: .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj
Comment (ralph-msft) [reply .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj:23]: We need to keep this change in for the Azure C# SDK repo since the Azure C# SDK repo has build step that tries to replace Nuget package references with project references. Otherwise we will continue to have build breakages due to the dependencies between System.ClientModel (in Azure repo which needs to build and publish first),  OpenAI (in separate repo which consumes SCM via Nuget package), and Azure OpenAI (in Azure repo which consumes OpenAI and SCM via Nuget packages).

Keeping here also makes it easier to move code between Jose's repo and the Azure C# SDK repo.

Comment (annelo-msft) [reply .dotnet.azure/sdk/openai/tools/TestFramework/src/OpenAI.TestFramework.csproj:23]: Cool, we can rip all of those out later once the upgrades go through - I have an issue tracking that.

File: .dotnet/tests/Assistants/AssistantTests.cs
Comment (ralph-msft) [reply .dotnet/tests/Assistants/AssistantTests.cs:1595]: I've updated this test (and the sync version that follows) to best represent the new code capabilities, and verified they both pass.

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (ralph-msft) [reply .dotnet/api/OpenAI.netstandard2.0.cs:55]: Unless those are related to the SCM 1.1.0 update, we should defer those changes to a different PR.

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (ralph-msft) [reply .dotnet/api/OpenAI.netstandard2.0.cs:7]: Let's defer these unrelated updates to a different PR.

File: .dotnet/api/OpenAI.netstandard2.0.cs
Comment (joseharriaga) [reply .dotnet/api/OpenAI.netstandard2.0.cs:55]: Sounds good

**Pull request merged**
